### PR TITLE
Only allow addons on evaluation shoots for versions >= 1.22

### LINKED ISSUE #4079: Only allow using Shoot addons for evaluation clusters beginning with v1.22
**How to categorize this issue?**

/area usability
/kind enhancement
/priority 3

**What would you like to be added**:
Only allow to enable addons in `.spec.addons` in `Shoot` resources if 
- `.spec.kubernetes.version` is lower than `v1.22.0`.
- `.spec.purpose=evaluation` and the `.spec.kubernetes.version` is greater or equal `v1.22.0`.

**Why is this needed**:
The addons are just provided by Gardener to enable "hello-world" use-cases and are not recommended for production usage.  They are not regularly maintained, updated without end-users' control, and don't allow for a lot of customization/configuration.
Hence, let's prohibit using them if the shoot purpose is not `evaluation` to make clear that end-users who want to use their clusters for production need to deploy their own addons.

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
Ensures that addons are only allowed on evaluation shoots if the Kubernetes version is >= 1.22.

**Which issue(s) this PR fixes**:
Fixes #4079

**Special notes for your reviewer**:

**Release note**:

```breaking user
Shoot addons are now only allowed on evaluation shoots if the Kubernetes version is >= 1.22.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Review by timebertt — DISMISSED** [review]
/lgtm

FYI: I changed the release note category to `breaking user` as I think this is the most appropriate.

File: pkg/apis/core/validation/shoot_test.go [hunk 1]
```diff
@@ -422,6 +422,34 @@ var _ = Describe("Shoot Validation Tests", func() {
...
+					Expect(errorList).To(BeEmpty())
+				} else {
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeForbidden),
+						"Field": Equal("spec.addons"),
+					}))))
+				}
+			},
+			Entry("should allow addons on evaluation shoots with version >= 1.22", core.ShootPurposeEvaluation, "1.22.0", true),
+			Entry("should forbid addons on testing shoots with version >= 1.22", core.ShootPurposeTesting, "1.22.0", false),
```
Comment (vpnachev) [inline pkg/apis/core/validation/shoot_test.go:444]: Could you add a test with pre-release k8s version, e.g `1.22.0-alpha.1`?

Comment (stoyanr) [reply pkg/apis/core/validation/shoot_test.go:444]: Done (in a new commit)
/squash

**Review by rfranzke — APPROVED** [review]

**Review by timebertt — APPROVED** [review]

**Pull request merged**